dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -746,6 +746,40 @@ QueryProcessingStage::Enum InterpreterSelectQuery::executeFetchColumns(Pipeline
                     "[InterpreterSelectQuery::executeFetchColumns] no region query",
                     ErrorCodes::LOGICAL_ERROR);
         }
+        else
+        {
+            if (auto managed_storage = std::dynamic_pointer_cast<IManageableStorage>(storage); managed_storage)
+            {
+                TableID table_id = managed_storage->getTableInfo().id;
+                // Only for (integration) test, because regions_query_info should never be empty if query is from TiDB or TiSpark.
+                // todo support partition table
+                auto & tmt = context.getTMTContext();
+                const auto regions = tmt.getRegionTable().getRegionsByTable(NullspaceID, table_id);
+                if (regions.empty())
+                {
+                    LOG_WARNING(
+                        log,
+                        "[InterpreterSelectQuery::executeFetchColumns] can not find any regions for the query, "
+                        "table_id={}",
+                        table_id);
+                }
+                else
+                {
+                    query_info.mvcc_query_info->regions_query_info.reserve(regions.size());
```

Comment:
Could we add a runtime check that `regions_query_info` must be empty?

Reply:
added